fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -10,26 +10,37 @@
 
 namespace fml {
 
-MessageLoopFuchsia::MessageLoopFuchsia() : running_(false) {
-  auto status = zx_timer_create(0, ZX_CLOCK_MONOTONIC, &timer_);
+MessageLoopFuchsia::MessageLoopFuchsia()
+  : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+    running_(false) {
+  auto status = zx_timer_create(ZX_TIMER_SLACK_LATE, ZX_CLOCK_MONOTONIC, &timer_);
+  FML_CHECK(status == ZX_OK);
+  timer_wait_ = new async::Wait(timer_, ZX_TIMER_SIGNALED);
+
+  auto wait_handler = [&](async_dispatcher_t* dispatcher,
+                        async::Wait* wait,
+                        zx_status_t status,
+                        const zx_packet_signal_t* signal
+                    ) {
+    RunExpiredTasksNow();
+    wait->Begin(loop_.dispatcher());
```

Comment:
Nit: Missing error handling on the `Begin` call. Maybe just assert `ZX_OK`.

Reply:
added a check but we also need to ensure we're still `running_` because otherwise it'll return `ZX_ERR_BAD_STATE` and we probably shouldn't be calling `Begin()` in that state. not sure of the perf implications of doing this.

Reply:
nvm; we can check for `status == ZX_ERR_CANCELED` to see if we're in the shutting-down state.